### ISSUE TITLE
Find another solution for editing the readme.md

### ISSUE THREAD

**Issue by dartcafe** [nextcloud/text#1171]
The current inplementation and display of the readme.md inside a folder has some confusing side effects. 

# Steps to reproduce

-  Don't have a Readme.md
-  click inside the "free space" above the file list
-  click anywhere outside the editor

# What happened
![grafik](https://user-images.githubusercontent.com/26707476/76640141-b6af8780-654f-11ea-8fa4-f7eb6ed97526.png)

# Expected behaviour
![grafik](https://user-images.githubusercontent.com/26707476/76640137-b0b9a680-654f-11ea-8e3f-e85f236f72e2.png)
- Clicking in a "free space" of the file list should not open an editor.
- No grown "white space"
- No magically appeared file in my folder. 

Some users got confused and complained about of the empty space and the fact, that an unknown and unwanted file appears in their folders.

Possible solution could be, to add a button for explicit adding/editing readme.md.

**Comment by dartcafe** [comment]
My thoughts about this: 
* Do not use an editable area, this confuses many users.
* Add a CTA which says "Add folder description" and let the user edit a new readme.md
* If a readme.md exists, 
  * show it's content in the current way (without useless whitespace) 
  * and add a CTA which says "edit folder description"

* Optionally remove empty readme.md, but this could be also surpringly to users.
* Optionally let the user edit the readme in a modal.

The handling of readme.md is not very common for 'normal' users.

**Comment by lars-sh** [comment]
I just created a Pull Request, which deletes the Readme.md file if its content should be set to an empty value and recreates a file if it does not exist, yet.

Warning: This Pull Request is not proven to work. It is a first try to help out on that problem. I knowing anything specific about Nextcloud development nor the Text app.

Does #1175 help to handle this problem?

**Comment by azul** [comment]
I think people are clicking this by accident because:
* the clickable area is too big
* there is no change in the cursor - so it seems like an okay place to click to say dismiss a menu somewhere else.

My understanding is that any click within the blue area will create the readme:
![Bildschirmfoto von 2022-01-05 18-37-20](https://user-images.githubusercontent.com/102902/148263095-0ff29047-366e-4bee-8810-704169116944.png)

Only the area around the text with a height of 44 px should be clickable and it should also change the cursor to a pointer to indicate clicking it will have an effect.

There's a lot more that could be done here. Right now the first click will create the readme... but it takes another click to edit it. The second click has to be more precise than the first. Making this a single click operation would be more smooth - but is probably not trivial.

If we really want to delay the creation of the Readme.md until there is some content we could create an empty `.Readme.md` that stays hidden by default and only when the user starts editing rename it to Readme.md. But as i said - a smaller clickable area and an indication that clicking there will have an effect would already go a long way.

**Comment by jancborchardt** [comment]
Agree with @azul’s proposals, small additions:
- Instead of a pointer, the curser could be a text caret `cursor: text` to communicate that it will create text
- The "single click flow" would indeed be very nice and smooth

**Comment by dartcafe** [comment]
> The "single click flow" would indeed be very nice and smooth

Again, the feedback, I revieve is the total opposite: I have 6 of 9 users complaining exactly about that on a regular basis, although they learned this behavior. 

That mostly seems to happen, if the browser is partly visible and overlapped by another application and a user clicks to the "empty" space to move the browser to the front. 

Actually this behavior is not expected by users.

**Comment by kobuki** [comment]
As much as I like NC and appreciate the hard work of the developers and support team involved, as an end user, I would simply expect an edit sign/icon, like a pencil to actually edit something, like in any other web based application in existence. Allowing users to actually create files and thus taking up resources (even if not that much), annoying them with even more empty space and unavoidable objects on top of the file list, based on random clicks on the existing empty space of a web page is simply nuts. It doesn't even take basic web ergonomics to consideration.

**Comment by LouisSung** [comment]
## TL;DR
Remove the **`,on:{click:t.createNew}`** in your **`nextcloud/data/apps/text/js/text-file.js`**
> Note: need NOT to restart/reconfig, unless extra cache mechanism like CDN is used

## How
The bound click event to create an empty *Readme.md* is the `@click="createNew"` in [**`src/views/RichWorkspace.vue#L25`**](https://github.com/nextcloud/text/blob/v23.0.3/src/views/RichWorkspace.vue#L25)
```html
<div v-if="showEmptyWorkspace" class="empty-workspace" @click="createNew">
```
which is then transpiled as `on:{click:t.createNew}}` in [**`js/text-files.js`**](https://github.com/nextcloud/text/blob/v23.0.3/js/text-files.js#L2)
```js
[t.showEmptyWorkspace?n("div",{staticClass:"empty-workspace",on:{click:t.createNew}}
```
### automation script
use `sed` to remove onclick event
```sh
# cd 'apps/text/js'
if cat text-files.js | grep -q '{staticClass:"empty-workspace",on:{click:t.createNew}}'; then
  echo 'disable Readme.md auto creation'
  sed -i -r 's/\{staticClass:"empty-workspace",on:\{click:t\.createNew\}\}/{staticClass:"empty-workspace"}/' text-files.js
else
  echo '@click:"createNew" event NOT found, check nextcloud/text#1171'
fi
```

see also: #586
https://help.nextcloud.com/t/disable-the-readme-editor-in-folder-header/73590
https://www.reddit.com/r/NextCloud/comments/fq4q8m/disable_blank_readmemd_creation/

**Comment by juliushaertl** [comment]
Let's close this as a duplicate of #3209 and https://github.com/nextcloud/text/issues/2803

**Comment by Oli72** [comment]
LouisSung's workaround https://github.com/nextcloud/text/issues/1171#issuecomment-1100890031 does not work for me in NC27 and NC28. `createNew` is not found.

**Comment by juliushaertl** [comment]
We won't support any hotpatched javascript asset guidelines, they are likely to break any time, but these days you can just disable the rich workspace with a config flag https://github.com/nextcloud/text?tab=readme-ov-file#configuration
